File: python/ohm.py
def ohm():
    while True:
        print("Calculadora de la Ley de Ohm")
        print("Selecciona la variable a calcular:")
        print("1. Voltaje (V)")
        print("2. Corriente (I)")
        print("3. Resistencia (R)")
        print("4. Salir")

        opcion = input("Ingresa tu opción (1, 2, 3 o 4): ")

        if opcion == "1":
            I = float(input("Ingresa la corriente (I) en amperios: "))
            R = float(input("Ingresa la resistencia (R) en ohmios: "))
            V = I * R
            print("El voltaje (V) es:", V, "voltios")
        elif opcion == "2":
            V = float(input("Ingresa el voltaje (V) en voltios: "))
            R = float(input("Ingresa la resistencia (R) en ohmios: "))
            if R != 0:
                I = V / R
                print("La corriente (I) es:", I, "amperios")
            else:
                print("La resistencia no puede ser 0.")
        elif opcion == "3":
            V = float(input("Ingresa el voltaje (V) en voltios: "))
            I = float(input("Ingresa la corriente (I) en amperios: "))
            if I != 0:
                R = V / I
                print("La resistencia (R) es:", R, "ohmios")
        elif opcion == "4":
            print("Gracias por usar el programa.")
            break
        else:
                print("La corriente no puede ser 0.")
    else:
            print("Opción inválida.")

ohm()

File: JavaScript/ohm.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function ohm() {
    function menu() {
    console.log("Calculadora de la Ley de Ohm");
    console.log("Selecciona la variable a calcular:");
    console.log("1. Voltaje (V)");
    console.log("2. Corriente (I)");
    console.log("3. Resistencia (R)");
    console.log("4. Salir");
    rl.question("Ingresa tu opción (1, 2, 3 o 4): ", (opcion) => {
        if (opcion === "1") {
        rl.question("Ingresa la corriente (I) en amperios: ", (Istr) => {
            const I = parseFloat(Istr);
            rl.question("Ingresa la resistencia (R) en ohmios: ", (Rstr) => {
            const R = parseFloat(Rstr);
            const V = I * R;
            console.log("El voltaje (V) es:", V, "voltios");
            menu();
            });
        });
        } else if (opcion === "2") {
        rl.question("Ingresa el voltaje (V) en voltios: ", (Vstr) => {
            const V = parseFloat(Vstr);
            rl.question("Ingresa la resistencia (R) en ohmios: ", (Rstr) => {
            const R = parseFloat(Rstr);
            if (R !== 0) {
                const I = V / R;
                console.log("La corriente (I) es:", I, "amperios");
            } else {
                console.log("La resistencia no puede ser 0.");
            }
            menu();
            });
        });
        } else if (opcion === "3") {
        rl.question("Ingresa el voltaje (V) en voltios: ", (Vstr) => {
            const V = parseFloat(Vstr);
            rl.question("Ingresa la corriente (I) en amperios: ", (Istr) => {
            const I = parseFloat(Istr);
            if (I !== 0) {
                const R = V / I;
                console.log("La resistencia (R) es:", R, "ohmios");
            } else {
                console.log("La corriente no puede ser 0.");
            }
            menu();
            });
        });
        } else if (opcion === "4") {
        console.log("Gracias por usar el programa.");
        rl.close();
        } else {
        console.log("Opción inválida.");
        menu();
        }
    });
    }
    menu();
}

ohm();
